packages/swing-store/src/hasher.js

```diff
@@ -3,8 +3,8 @@ import { assert } from '@agoric/assert';
 import { createHash } from 'crypto';
 
 /**
- * @typedef { (initial?: string) => {
- *             add: (more: string) => void,
+ * @typedef { (initial?: string | Buffer) => {
+ *             add: (more: string | Buffer) => void,
```

Comment:
Not an answer, more of a principle, but in general I'm nervous about how easy it is for us to assume that everything is a string, rather than properly handling binary blobs. There's certainly a debugging benefit to computing a tree of hashes by hashing double-sized hex strings, rather than the direct (binary/Buffer/Uint8Array) output of the hasher. But from a protocol design POV, it's awkward to tell implementors to convert the hash output to hex before passing it into another hash function.. it makes it look like we forgot how to use binary data.

Reply:
I believe what happened was that at one point in the evolution of the code something _was_ (legitimately) adding a Buffer, which caused TypeScript to whine and forced a change to the type declaration, and then with further evolution that use of Buffer went away again, but nothing pushed the type declaration to be changed back.